### ISSUE TITLE
Use native PowerBI Visual Tooltips

### ISSUE THREAD

**Issue by ryanbaumann** [mapbox/mapboxgl-powerbi#52]
# Problem

Tooltips in the current master release are scaled based on the window size of the visualization.  This means tooltips are

1. Squished visually based on the size of the viz container on the users screen. 
    * Popups are not very legible on mobile or small viz containers on web.
2. Data for tooltips does not need to be copied to the geojson source passed to the map as a viz, reducing the size of geojson tiles and improving map viz layer load time performance.

- PowerBI Visual tooltip tutorial: https://github.com/Microsoft/PowerBI-visuals/blob/master/Visual/Tooltips.md

# Deliverable

- Update PowerBI Mapboxgl custom viz to use PowerBI tooltips instead of HTML / CSS elements in the map viz container.

@szilardhuber @endre-jenei @baspeti @petispaespea

**Comment by otravers** [comment]
Note for the record: the Tooltip field that allows the loading of a report as a tooltip is not (yet?) available for custom visuals.
https://docs.microsoft.com/en-us/power-bi/desktop-tooltips#limitations-and-considerations

**Comment by ryanbaumann** [comment]
Great note @otravers - yeah, tooltip reports is a great new feature that wouldn't be unlocked by this work changing to native tooltips.

**Comment by ryanbaumann** [comment]
The initial version of native tooltips are merged in!

@szilardhuber tracking performance questions with the native tooltip update:

[Video
](https://www.dropbox.com/s/f8ntn1cfp3hmlhs/Power%20BI%20-%20Google%20Chrome%203_28_2018%208_38_15%20AM.mp4?dl=0)

TL;DR native tooltips are slow to update and are taking a bunch of main thread CPU time.  Possible things to check:

1. Reduce `buffer` pixel size of query rendered features to 1-2 pixels: https://github.com/mapbox/mapboxgl-powerbi/pull/85/files#diff-6ec3206179d5be96d1f1bdcf47f1dfebR72
2. Only display the top 3 features in the tooltip - currently, it looks like there is no limit that will display.
3. Change the mouse to a `pointer` when a tooltip will appear on the map canvas

**Comment by ryanbaumann** [comment]
Closed in https://github.com/mapbox/mapboxgl-powerbi/pull/113 and https://github.com/mapbox/mapboxgl-powerbi/pull/108 and https://github.com/mapbox/mapboxgl-powerbi/pull/85

**Comment by ryanbaumann** [comment]
> Change the mouse to a pointer when a tooltip will appear on the map canvas

@szilardhuber @baspeti can we change the mouse cursor to a pointer in the current beta when popups appear?

**Comment by szilardhuber** [comment]
@ryanbaumann https://github.com/mapbox/mapboxgl-powerbi/pull/115
